integration-cli/docker_cli_plugins_test.go

```diff
@@ -251,3 +251,30 @@ func (s *DockerSuite) TestPluginInspect(c *check.C) {
 	_, _, err = dockerCmdWithError("plugin", "inspect", "-f", "{{.Id}}", id[:5])
 	c.Assert(err, checker.NotNil)
 }
+
+func (s *DockerSuite) TestPluginUpgrade(c *check.C) {
+	testRequires(c, DaemonIsLinux, Network, SameHostDaemon, IsAmd64)
+
+	dockerCmd(c, "plugin", "install", "--grant-all-permissions", pNameWithTag)
+	out, _, err := dockerCmdWithError("plugin", "upgrade", "--grant-all-permissions", pNameWithTag)
+	c.Assert(err, checker.NotNil, check.Commentf(out))
+	c.Assert(out, checker.Contains, "disabled before upgrading")
+
+	out, _ = dockerCmd(c, "plugin", "inspect", "--format={{.ID}}", pNameWithTag)
+	id := strings.TrimSpace(out)
+	stat, err := os.Stat(filepath.Join(dockerBasePath, "plugins", id, "rootfs"))
+	c.Assert(err, checker.IsNil)
+	modTime := stat.ModTime()
+
+	dockerCmd(c, "plugin", "disable", pNameWithTag)
+	dockerCmd(c, "plugin", "upgrade", "--grant-all-permissions", pNameWithTag)
+
+	stat, err = os.Stat(filepath.Join(dockerBasePath, "plugins", id, "rootfs"))
```

Comment:
A more effective test would be to create volume before and check for existence after upgrade.